py/client/pydeephaven/session.py

```diff
@@ -384,9 +384,10 @@ def run_script(self, script: str) -> None:
         Raises:
             DHError
         """
-        response = self.console_service.run_script(script)
-        if response.error_message != '':
-            raise DHError("could not run script: " + response.error_message)
+        with self._r_lock:
```

Comment:
I'm a bit confused about the current design.
1. `_console_service.start_console` does the relevant locking in `_console_services`.
2. If `_console_service.run_script` requires locking, what is that locking not in `_console_services`?
This decision just feels inconsistent.